### PR TITLE
ProjectSettings: enhance add action error, fix confirm dialog

### PR DESCRIPTION
**Pull request by ghost**
- Disable add action button if name is invalid

image
image

(Old screenshot: [1](https://user-images.githubusercontent.com/26753932/29988841-888e8934-8f99-11e7-84c2-9915df879351.png), [2](https://user-images.githubusercontent.com/26753932/29988840-888b9ecc-8f99-11e7-8760-5642db4636fc.png))

~~Want some opinion on this though. When a valid action is added, it'll say that it's already exists because it has just been successfully added and the LineEdit is not cleared. Would it be misleading as error? Should the LineEdit be cleared, text/color be changed, or this entire behavior be removed and use the existing popup instead?~~ This is not the case anymore in latest commit.

- Fix dialog being ConfirmationDialog instead of AcceptDialog

### CODE REVIEW

**Comment by neikeq** [comment]
The wording for the messages is not the ideal IMO

File: editor/project_settings_editor.cpp [hunk 1]
```diff
@@ -809,28 +811,41 @@ void ProjectSettingsEditor::_item_del() {
...
-		message->set_text(TTR("Invalid action (anything goes but '/' or ':')."));
-		message->popup_centered(Size2(300, 100) * EDSCALE);
-		return;
+		if (p_action.find("/") != -1 || p_action.find(":") != -1) {
+			action_add->set_text(TTR("Can't contain '/' or ':'."));
+			action_add->set_disabled(true);
+			return;
+		}
+		if (ProjectSettings::get_singleton()->has("input/" + p_action)) {
+			action_add->set_text(TTR("Added"));
```
Comment (akien-mga) [inline editor/project_settings_editor.cpp]: "Already existing" would be clearer no?

Comment (Noshyaar) [reply editor/project_settings_editor.cpp]: I'm not sure. For me, if I type in an action name, press enter and it says "Already existing", I'd think I've added it before.

Comment (akien-mga) [reply editor/project_settings_editor.cpp]: Well that's what it means no? :)

Comment (akien-mga) [reply editor/project_settings_editor.cpp]: Once you've added an action, `action_add` should be cleared, so you don't need to handle this case. Then if you test for the action already existing, that's what the error should say.

**Comment by akien-mga** [comment]
Merged while we hadn't finished discussing it, see http://godot.eska.me/irc-logs/devel/2017-09-04.log
@reduz mentions that we usually use a colored label to show such warnings or errors instead of changing the button.

**Pull request merged**